Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -480,6 +480,13 @@ def get_matrix_2d_ragged(workspace, normalize_by_bin_width, histogram2D=False, t
             max_value = max(max_value, xtmp.max())
             diff = xtmp[1:] - xtmp[:-1]
             delta = min(delta, diff.min())
+    xtmp = workspace.readX(0)
+    if delta == np.finfo(np.float64).max:
+        delta = (xtmp[1:] - xtmp[:-1]).min()
```

Comment:
Can we use `numpy.diff().min()` here (and on line 481)? I think it signals the intent of finding the difference a little clearer then the array index trick.